File: src/main/java/org/goafabric/personservice/controller/dto/Person.java
package org.goafabric.personservice.controller.dto;

import jakarta.annotation.Nullable;
import jakarta.validation.constraints.NotNull;
import jakarta.validation.constraints.Size;

import java.util.List;

public record Person (
    @Nullable String id,
    @Nullable Long version,
    @NotNull @Size(min = 3, max = 255) String firstName,
    @NotNull @Size(min = 3, max = 255) String lastName,
    List<Address> address) {
}

File: src/main/kotlin/org/goafabric/personservice/controller/dto/Person.kt
package org.goafabric.personservice.controller.dto

import jakarta.validation.constraints.Size

data class Person (
    val id:  String? = null,
    val version:  Long? = null,
    @field:Size(min = 3, max = 255) val firstName: String,
    @field:Size(min = 3, max = 255) val lastName: String,
    val address: List<Address>
)